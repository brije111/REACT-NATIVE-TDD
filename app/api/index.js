import Axios from 'axios';

export default Axios.create({baseURL: 'my_base_url'});
