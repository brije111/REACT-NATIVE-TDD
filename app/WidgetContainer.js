import React, {useState, useEffect} from 'react';
import {Text, View} from 'react-native';
import api from './api';
import Hello from './Hello';

const WidgetContainer = () => {
  const [widgets, setWidgets] = useState([]);

  useEffect(() => {
    console.log('before response');
    api.get('/widgets').then((response) => {
      console.log('after response');
      setWidgets(response.data);
    });
  }, []);

  return (
    <View>
      <Hello name="Hello World" />
      {widgets.map((widget) => (
        <Text key={widget.id}>{widget.name}</Text>
      ))}
    </View>
  );
};

export default WidgetContainer;
