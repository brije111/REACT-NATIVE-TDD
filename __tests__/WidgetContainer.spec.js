import React from 'react';
import {render} from 'react-native-testing-library';
import WidgetContainer from '../app/WidgetContainer';
import api from '../app/api';
//import act from '@testing-library/react';
jest.mock('../app/api');

describe('WidgetContainer', () => {
  it('loads widgets upon mount', async () => {
    api.get.mockResolvedValue({
      data: [
        {id: 1, name: 'Widget 1'},
        {id: 2, name: 'Widget 2'},
      ],
    });
    const {findByText} = render(<WidgetContainer />);
    //debug();
    //expect(queryByText('Widget 1')).not.toBeNull();
    //expect(queryByText('Widget 2')).not.toBeNull();
    await findByText('Widget 1');
    await findByText('Widget 2');
  });

  // it('should match with the snapshot', () => {
  //   const tree = render(<WidgetContainer />).toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
