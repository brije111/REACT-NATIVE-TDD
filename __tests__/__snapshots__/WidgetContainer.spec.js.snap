// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetContainer should match with the snapshot 1`] = `
<View>
  <Text>
    Hello, 
    Hello World
    !
  </Text>
</View>
`;
